base/linalg/cholesky.jl

```diff
@@ -324,11 +324,13 @@ convert{T}(::Type{CholeskyPivoted{T}},C::CholeskyPivoted) =
     CholeskyPivoted(AbstractMatrix{T}(C.factors),C.uplo,C.piv,C.rank,C.tol,C.info)
 convert{T}(::Type{Factorization{T}}, C::CholeskyPivoted) = convert(CholeskyPivoted{T}, C)
 
-full{T,S}(C::Cholesky{T,S}) = C.uplo == 'U' ? C[:U]'C[:U] : C[:L]*C[:L]'
-function full(F::CholeskyPivoted)
-    ip = invperm(F[:p])
-    return (F[:L] * F[:U])[ip,ip]
-end
+convert(::Type{Matrix}, C::Cholesky) = C.uplo == 'U' ? C[:U]'C[:U] : C[:L]*C[:L]'
```

Comment:
For each array-parametic type that originally had a `full` definition not necessarily returning an `Array` (e.g. that you highlighted, and the array-parametric factorization types generally), I've added a wrapping `convert(Array, )` to ensure the return will now be an `Array`.

Ultimately I think the pattern for `LDLt`
```julia
convert(::Type{SymTridiagonal}, F::LDLt) = ...
convert(::Type{AbstractMatrix}, F::LDLt) = convert(SymTridiagonal, F)
convert(::Type{AbstractArray}, F::LDLt) = convert(AbstractMatrix, F)
convert(::Type{Matrix}, F::LDLt) = convert(Array, convert(AbstractArray, F))
convert(::Type{Array}, F::LDLt) = convert(Matrix, F)
full(F::LDLt) = convert(Array, F) 
```
is where the methods for these cases should go, solving #12153. Another PR though. Thanks!